cranelift/codegen/src/isa/s390x/inst/emit.rs

```diff
@@ -1426,7 +1426,12 @@ impl MachInstEmit for Inst {
                     _ => unreachable!(),
                 };
                 if have_rr && rd.to_reg() == rn {
-                    let inst = Inst::AluRR { alu_op, rd, rm };
+                    let inst = Inst::AluRR {
+                        alu_op,
+                        rd,
+                        ri: rd.to_reg(),
```

Comment:
Updated to use `rn`'s value, thanks. I actually lean slightly toward keeping these fields named `ri`, to make it clear that they are artificial instruction fields, the "input" side of the dest reg, rather than a real `rn` field (this also made it easier to grep for things when updating tests just now!). But I'm happy to alter the field name as well if you feel strongly about this.